### PR TITLE
Install node with n

### PR DESCRIPTION
**Pull request by agarzola**
## Description

This PR replaces `php-apache-nvm` with a much simpler `php-apache-n`. This new image installs [n](https://github.com/tj/n) in `/usr/local/bin/n`, installs Node.js LTS (and npm) with `n lts`, and then uses the newly-installed npm to install Yarn globally.

## Motivation / Context

The nvm-based solution was too hacky due to nvm’s implementation as a shell function that must be sourced in a given session. This made it so that any commands that involved Node.js (i.e. front-end tasks, functional tests, etc) had to be either run in the context of a Bash login session or prefixed with `. ~/.bashrc` to ensure the nvm function and whatever version of Node is currently in use _with_ nvm were available before running the command. In this way, nvm acts as a sort of middleman because it must be sourced in any session that intends to use Node.js even if nvm itself is not invoked directly, which is both restrictive and cumbersome.

n is different. It does not require being sourced in any given session. Instead, it can be installed like any shell script and run as needed. It still uses Bash, but it uses a shebang at the top of its script file to indicate this, meaning it can be simply invoked in any system that _has_ Bash available. In addition to this, n installs Node.js in a global location, so that it is available from within any session without n itself acting as a middleman.

## How This Has Been Tested

First, I added a temporary build job to this branch to create a `php-apache-n` image and [publish it to DockerHub](https://hub.docker.com/r/chromatichq/php-apache-n/tags). Then I created [this Draft PR in CHQ](https://github.com/ChromaticHQ/chromatichq.com/pull/3649), which modifies our Tugboat config to use this new Docker image and updates the build commands to use `n auto`. `n auto` installs the version of Node.js specified in `.nvmrc` (n supports a variety of files to specify a Node.js version). The PR also removes all `bash -lc` nonsense from our Node.js-based commands. Finally, it also merges the changes in [this Dependabot PR](https://github.com/ChromaticHQ/chromatichq.com/pull/3583) that updates `chq-robo` to v2.11.0, which removes the `bash -lc` prefix from our theme build commands.

The fact that the Draft CHQ PR noted above builds in Tugboat without any issues should sufficiently demonstrate that the changes in this PR work as intended. Specifically, it proves the following:
- `n` commands work as expected without the need to source it or run commands in a special context of any kind.
- The Yarn installation in the new `php-apache-n` works as expected, as the Yarn commands in the PR run w/o issue.
- The version of Node.js specified in CHQ’s `.nvmrc` file (14) is detected and installed.

See screenshots below.

## Screenshots

|image|
|---|
|`n auto` command running without any special scripts, sourcing, or special contexts. It detects `14` in our `.nvmrc`, and installs it.|

|image|
|---|
|Yarn is available to our theme build commands, despite that branch using the version of `chq-robo` that removes the `bash -lc` prefix.|

|image|
|---|
|The Tugboat environment’s `apache-php7` service shows the expected version of Node.js is installed in a global location available to any session.|

## Documentation

I updated the README.

### CODE REVIEW

File: services/php-apache-n/Dockerfile
Comment (apotek) [inline services/php-apache-n/Dockerfile:9]: This way of installing `n` gives me the heeby jeebies. If it is the official way, so be it, and disregard this comment, but I would much rather we were tying the installation to a tagged release or release bundle rather than whatever happens to be latest master.

**Review by apotek — APPROVED** [review]
Assuming this is an agreed upon change from nvm to n, then this looks good, though I did question a detail in the methodology of the n install. Nice work.

File: services/php-apache-n/Dockerfile
Comment (agarzola) [reply services/php-apache-n/Dockerfile:9]: Good call! Looks like Git’s `clone` command [accepts a `--branch` option](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---branchltnamegt) which can accept a tag, so something like this _should_ work:
```sh
git clone --branch v8.0.1 https://github.com/tj/n
```

That would be one way to “lock” it, although technically the maintainer could rewrite the tag to point to any commit in the future, but that’s highly unlikely.

I’ll give that a shot and confirm whether it works…

File: README.md
Comment (adamzimmermann) [inline README.md:12]: This readme section cleared up so much for me. I was struggling to wrap my head around the big picture. 🏆

**Review by adamzimmermann — APPROVED** [review]
I like it.

File: services/php-apache-n/Dockerfile
Comment (agarzola) [reply services/php-apache-n/Dockerfile:9]: Done! I rebuilt `php-apache-n` after that and then rebuilt the Tugboat preview for the related CHQ PR. That Tugboat preview is past the installation of Node v14 now, so it looks like this works as expected.

I’m keeping this as a way to lock our version of n to v8.0.1, with the caveat that the tag is not exactly immutable, but at least we’re protected from any bugs or breaking changes that might get introduced into the project’s HEAD. Thanks for the suggestion, @apotek!

See 4e9911c.

**Comment by agarzola** [comment]
After [consulting with the team](https://chromatic.slack.com/archives/C02K4QA7V/p1639070160039800), I renamed `php-apache-n` to `php-apache-node`. Once checks pass, I’ll merge this PR and update the CHQ PR.

File: services/php-apache-n/Dockerfile
Comment (apotek) [reply services/php-apache-n/Dockerfile]: The `--branch` method is probably good enough. Another option would be to checkout a specific commit. But this reads more clearly, intent-wise. If you checked out a specific commit, someone would come along later and wonder huh/why? Whereas this documents your intent. Nice stuff.

**Pull request merged**
